### PR TITLE
Linter enhancements part 2

### PR DESCRIPTION
**Pull request by rachidlamouri**
## Description

**Note:** there is additional reasoning in some of the commit messages

I realized that airbnb-typescript does not extend the recommended typescript eslint rules, which is why the rules that catch common mistakes weren't already enabled. I extended the recommended ruleset and replaced the redundant custom rules.

I also updated prettier to add trailing commas every. EVERYWHERE. Every line. So many commas. CommaScript 2022®.

## Ticket

https://github.com/normal-devs/klicker-knight/projects/1#card-74538171

### CODE REVIEW

**Comment by rachidlamouri** [comment]
- [ ] Make sure the rule for using `Record<K, V>` instead of `object` is in place

File: src/klickerKnight.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import { Command } from '@oclif/command';
 
 export class KlickerKnight extends Command {
-  async run() {
+  async run(): Promise<void> {
```
Comment (jshemas) [inline src/klickerKnight.ts:4]: 👌

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/.nyc_output/
+/lib/
+/node_modules/
+!.mocharc.js
```
Comment (jshemas) [inline .eslintignore:4]: Is this needed? Are these files ignored by default?

**Review by jshemas — APPROVED** [review]

File: .eslintignore (hunk 2 above)
Comment (rachidlamouri) [reply .eslintignore:4]: Yes they are ignored by default

**Review by jshemas — APPROVED** [review]

**Pull request merged**